### PR TITLE
Feature/auth/role permissions

### PR DESCRIPTION
**Pull request by zingmane**
... many TODOs
this PR contains a first end to end workflow for delete table authorization

### CODE REVIEW

File: src/main/scala/com/campudus/tableaux/router/auth/permission.scala [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+case object ScopeCell extends Scope { override val name = "cell" }
+case object ScopeMedia extends Scope { override val name = "media" }
+
+object ConditionContainer {
+
+  def apply(jsonObjectOrNull: JsonObject): ConditionContainer = {
+
+    val jsonObject: JsonObject = Option(jsonObjectOrNull).getOrElse(Json.emptyObj())
+
+    val ct: ConditionOption = Option(jsonObject.getJsonObject("table")).map(ConditionTable).getOrElse(NoneCondition)
```
Comment (Zwergal) [inline src/main/scala/com/campudus/tableaux/router/auth/permission.scala]: Please don't use this short variable names. It's not readable for other people. Please name them like conditionTable = Option... and so on.

File: src/main/scala/com/campudus/tableaux/router/auth/permission.scala [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+                              conditionLangtag: ConditionOption) {}
+
+abstract class ConditionOption(jsonObject: JsonObject) {
+  val conditionMap: Map[String, String] = toMap(jsonObject)
+
+  protected def toMap(jsonObject: JsonObject) = {
+    jsonObject.asMap.toMap.asInstanceOf[Map[String, String]]
+  }
+
+  // TODO remove ugly method overloading, but how?
```
Comment (Zwergal) [inline src/main/scala/com/campudus/tableaux/router/auth/permission.scala]: isMatching accepts parameter of any type and inside the method you match for the type and act regardingly.
Not the best solution either...

File: src/main/scala/com/campudus/tableaux/router/auth/permission.scala [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+  override def toString: String =
+    role2permissions
+      .map({
+        case (key, permission) => s"$key => ${permission.toString}"
+      })
+      .mkString("\n")
+
+  // TODO possibly obsolete
+  def getPermissionsForRoles(roleNames: Seq[String]): Seq[Permission] =
+    role2permissions.filter({ case (k, _) => roleNames.contains(k) }).values.flatten.toSeq
```
Comment (Zwergal) [inline src/main/scala/com/campudus/tableaux/router/auth/permission.scala]: Please don't use this short variable names. It's not readable for other people. Please name them like `key`

File: src/main/scala/com/campudus/tableaux/router/auth/permission.scala [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+    *
+    * @return a subset of permissions
+    */
+  def filterPermissions(roleNames: Seq[String],
+                        permissionType: PermissionType,
+                        action: Action,
+                        scope: Scope): Seq[Permission] = {
+
+    val permissions: Seq[Permission] =
+      role2permissions.filter({ case (k, _) => roleNames.contains(k) }).values.flatten.toSeq
```
Comment (Zwergal) [inline src/main/scala/com/campudus/tableaux/router/auth/permission.scala]: Please don't use this short variable names. It's not readable for other people. Please name them like `key`

File: src/main/scala/com/campudus/tableaux/router/auth/permission.scala (hunk 2 above)
Comment (zingmane) [reply src/main/scala/com/campudus/tableaux/router/auth/permission.scala]: solved with a container

File: src/main/scala/com/campudus/tableaux/router/auth/permission.scala [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+
+import com.campudus.tableaux.UnauthorizedException
+import com.campudus.tableaux.database.domain.{ColumnType, Table}
+import com.campudus.tableaux.helper.JsonUtils.asSeqOf
+import org.vertx.scala.core.json._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+
+object RoleModel {
```
Comment (Zwergal) [inline src/main/scala/com/campudus/tableaux/router/auth/permission.scala]: Here are a lot of classes, case classes, objects, companion objects, sealed traits etc. in one .scala file.
Wouldn't it be better to split this up in separate files?

**Pull request merged**